src/main/java/jenkins/scm/api/SCMSource.java

```diff
@@ -261,6 +262,52 @@ protected SCMRevision retrieve(@NonNull SCMHead head, @NonNull TaskListener list
     }
 
     /**
+     * Looks up a specific revision based on some SCM-specific set of permissible syntaxes.
+     * Delegates to {@link #retrieve(String, TaskListener)}.
+     * @param revision might be a branch name, a tag name, a cryptographic hash, a revision number, etc.
+     * @param listener the task listener (optional)
+     * @return a valid revision object corresponding to the argument, with a usable corresponding head, or null if malformed or not found
+     * @throws IOException if an error occurs while performing the operation.
+     * @throws InterruptedException if any thread has interrupted the current thread.
+     * @since FIXME
+     */
+    @CheckForNull
+    public final SCMRevision fetch(@NonNull String revision, @CheckForNull TaskListener listener)
+            throws IOException, InterruptedException {
```

Comment:
The overload taking a `String[]` bypasses the shell’s parser, so it will wind up doing the same thing as

```
$ git rev-parse '|| sudo cat /etc/shadow^commit'
|| sudo cat /etc/shadow^commit
fatal: ambiguous argument '|| sudo cat /etc/shadow^commit': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```